core/kazoo_json/src/kz_json.erl

```diff
@@ -402,6 +406,22 @@ sum(?JSON_WRAPPER(_)=JObj1, Value, Sumer, Keys)
     V = get_value(Syek, JObj1),
     set_value(Syek, Sumer(V, Value), JObj1).
 
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Reorder JSON objects according to the given list of binaries.
+%% Given Path MUST resolve to all distinct values in the given objects.
+%% These resolved values MUST all be in the list of binaries too.
+%% List of binaries MUST NOT contain duplicates.
+%% Both lists MUST be of same size.
+%% @end
+%%--------------------------------------------------------------------
+-spec order_by(path(), ne_binaries(), objects()) -> objects().
+order_by(Path, Ids, JObjs)
```

Comment:
I'm confused why you want/need a list of IDs? Shouldn't you order the JSON objects by the path?

Also, process dictionary seems unnecessary.

Untested but what about:

```erlang
order_by(Path, JObjs) ->
    WithValues = [{get_value(Path, JObj), JObj} || JObj <- JObjs],
    [JObj || {_, JObj} <- lists:keysort(1, WithValues)].
```

Does this do the same job?